src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
```

Comment:
Unchecked cast: 'java.lang.Object[]' to 'T[]'  can be potentially dangerous because it cannot be checked by the compiler for the correctness of all operations
```suggestion
    private Object[] array = new Object[def_capacity];
```